gtfs-validator/main/src/test/java/org/mobilitydata/gtfsvalidator/validator/StopTooFarFromTripShapeValidatorTest.java

```diff
@@ -0,0 +1,164 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.validator;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import org.mobilitydata.gtfsvalidator.table.GtfsStop;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsLocationType;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTime;
+import org.mobilitydata.gtfsvalidator.table.GtfsStopTimeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsShape;
+import org.mobilitydata.gtfsvalidator.table.GtfsShapeTableContainer;
+import org.mobilitydata.gtfsvalidator.table.GtfsTrip;
+import org.mobilitydata.gtfsvalidator.table.GtfsTripTableContainer;
+import org.mobilitydata.gtfsvalidator.notice.NoticeContainer;
+import org.mobilitydata.gtfsvalidator.notice.StopTooFarFromTripShapeNotice;
+
+import static com.google.common.truth.Truth.assertThat;
+
+@RunWith(JUnit4.class)
+public class StopTooFarFromTripShapeValidatorTest {
+
+    public static GtfsStop createStop(long csvRowNumber, String stopId, double stopLat, double stopLon, GtfsLocationType locationType) {
+        return new GtfsStop.Builder().setCsvRowNumber(csvRowNumber).setStopId(stopId).setStopLat(stopLat).setStopLon(stopLon).setLocationType(locationType.getNumber()).build();
+    }
+
+    public static GtfsStopTime createStopTime(long csvRowNumber, String tripId, String stopId, int stopSequence) {
+        return new GtfsStopTime.Builder().setCsvRowNumber(csvRowNumber).setTripId(tripId).setStopId(stopId).setStopSequence(stopSequence).build();
+    }
+
+    public static GtfsShape createShape(long csvRowNumber, String shapeId, double shapePtLat, double shapePtLon, int shapePtSequence) {
+        return new GtfsShape.Builder().setCsvRowNumber(csvRowNumber).setShapeId(shapeId).setShapePtLat(shapePtLat).setShapePtLon(shapePtLon).setShapePtSequence(shapePtSequence).build();
+    }
+
+    public static GtfsTrip createTrip(long csvRowNumber, String tripId, String shapeId) {
+        return new GtfsTrip.Builder().setCsvRowNumber(csvRowNumber).setTripId(tripId).setShapeId(shapeId).build();
+    }
+
+    private static GtfsStopTableContainer createStopTable(String[] stopIds, double[] stopLats, double[] stopLons, GtfsLocationType[] locationTypes, NoticeContainer noticeContainer) {
+        Preconditions.checkArgument(stopIds.length == stopLats.length, "stopIds.length must be equal to stopLats.length");
+        Preconditions.checkArgument(stopLats.length == stopLons.length, "stopLats.length must be equal to stopLons.length");
+        Preconditions.checkArgument(stopIds.length == locationTypes.length, "stopIds.length must be equal to locationTypes.length");
+        ArrayList<GtfsStop> stops = new ArrayList<>();
+        stops.ensureCapacity(stopIds.length);
```

Comment:
no need to call ensureCapacity

Reply:
Sure. I have deleted all "ensureCapacity" checks in the file.